components/dialog/dialog-mixin.js

```diff
@@ -0,0 +1,171 @@
+import { html } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+export const DialogMixin = superclass => class extends RtlMixin(superclass) {
+
+	static get properties() {
+		return {
+			opened: { type: Boolean, reflect: true },
+			title: { type: String },
+			_height: { type: Number },
+			_overflowBottom: { type: Boolean },
+			_overflowTop: { type: Boolean },
+			_state: { type: String, reflect: true },
+			_width: { type: Number },
+		};
+	}
+
+	constructor() {
+		super();
+		this.opened = false;
+		this._height = 0;
+		this._width = 0;
+		this._hasNativeDialog = (window.HTMLDialogElement !== undefined);
+		//this._hasNativeDialog = false;
+		this._updateSize = this._updateSize.bind(this);
+		this._updateOverflow = this._updateOverflow.bind(this);
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+		if (name === 'opened' && oldval !== newval) {
+			if (this.opened) this._open();
+			else this._close();
+		}
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.addEventListener('keydown', this._handleKeyDown);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this.removeEventListener('keydown', this._handleKeyDown);
+	}
+
+	_close() {
+		if (!this._state) return;
+		const dialog = this.shadowRoot.querySelector('.d2l-dialog-outer');
+		const transitionEnd = () => {
+			dialog.removeEventListener('transitionend', transitionEnd);
+			dialog.querySelector('.d2l-dialog-content').removeEventListener('scroll', this._updateOverflow);
+			window.removeEventListener('resize', this._updateSize);
+			if (this._hasNativeDialog) {
+				dialog.close();
+			}
+			this._state = null;
+			this.opened = false;
+		};
+		dialog.addEventListener('transitionend', transitionEnd);
+		this._state = 'hiding';
+	}
+
+	_getHeight() {
+		const availableHeight = window.innerHeight - 130;
```

Comment:
Nit: `130` could go in a constant somewhere... similar to the `60` down below.

Reply:
Will do.